gamemodes/terrortown/gamemode/server/sv_network_sync.lua

```diff
@@ -0,0 +1,677 @@
+			initialized_clients[#initialized_clients + 1] = receiver
+		end
+	end
+end
+
+---
+-- This is the callback for clients requesting a full state update.
+-- It will just send a full state update to the client that requested it.
+--
+-- @param number len The length of the network message
+-- @param Player client the client that made the request
+-- @internal
+local function ClientRequestFullStateUpdate(len, client)
+	TTT2NET:SendFullStateUpdate(client)
+end
+net.Receive(TTT2NET.NETMSG_REQUEST_FULL_STATE_UPDATE, ClientRequestFullStateUpdate)
+
+---
+-- This is used to write a path table to the current network message.
+--
+-- @param table path The path table to send
+function TTT2NET:NetWritePath(path)
+	net.WriteString(pon.encode(path))
+end
+
+---
+-- This is used to write a metadata table to the current network message.
+--
+-- @param table metadata The metadata to send
+function TTT2NET:NetWriteMetaData(metadata)
+	net.WriteBool(metadata == nil)
+	if metadata == nil then
+		return
+	end
+
+	net.WriteString(metadata.type)
+	if metadata.type == "int" then
+		net.WriteUInt(metadata.bits, 6) -- max 32 bits
+		net.WriteBool(metadata.unsigned)
+	end
+end
+
+---
+-- This is used to write a value to the current network message,
+-- based on the given metadata.
+-- @note Nil values are preserved and can be "sent".
+--
+-- @param table metadata The metadata for the given value
+-- @param any|nil val The value to send, can also be nil
+function TTT2NET:NetWriteData(metadata, val)
+	-- Check if the value is nil, to also allow setting a value to nil
+	net.WriteBool(val == nil)
+	if val == nil then
+		return
+	end
+
+	-- Decide on how to send the data, or fallback to string
+	if metadata.type == "int" then
+		if metadata.unsigned then
+			net.WriteUInt(val, metadata.bits or 32)
+		else
+			net.WriteInt(val, metadata.bits or 32)
+		end
+	elseif metadata.type == "bool" then
+		net.WriteBool(val)
+	elseif metadata.type == "float" then
+		net.WriteFloat(val)
+	else
+		net.WriteString(val)
+	end
+end
+
+---
+-- Player extensions
+-- Some simple functions on the player class, to simplify the use of this system.
+-- They will automatically set the metadata.
+
+local plymeta = assert(FindMetaTable("Player"), "[TTT2NET] FAILED TO FIND PLAYER TABLE")
+
+---
+-- Sets a bool value at the given path on the player.
+--
+-- @param any|table path The path to set the value for
+-- @param bool|nil value The value to set
+function plymeta:TTT2NETSetBool(path, value)
+	TTT2NET:SetOnPlayer(path, { type = "bool" }, value, self)
+end
+
+---
+-- Sets an int value at the given path on the player.
+--
+-- @param any|table path The path to set the value for
+-- @param int|nil value The value to set
+-- @param int|nil bits The bits that this int needs to be stored (optional, otherwise a default of 32 is used)
+function plymeta:TTT2NETSetInt(path, value, bits)
+	TTT2NET:SetOnPlayer(path, { type = "int", bits = bits }, value, self)
```

Comment:
CS: table with two entries on initialization in different lines. Should match up with the CS anyways